controllers/discovery/initDiscoveryUserData.js

```diff
@@ -30,6 +31,11 @@ const InitDiscoveryUserData = async (req, res) => {
       .join(',')})`;
   }
 
+  let filterExcludedUser = '';
+  if (excluded_users && excluded_users?.length > 0) {
+    filterExcludedUser += `AND A.user_id NOT IN (:excluded_users)`;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `excluded_users` and the creation of the `filterExcludedUser` string are done correctly. However, it's important to validate the length of `excluded_users` against `MAX_EXCLUDED_USERS` here as well, to ensure that the SQL query does not become too large if a huge list of excluded users is provided.
```diff
- if (excluded_users && excluded_users?.length > 0) {
+ if (excluded_users && excluded_users?.length > 0 && excluded_users.length <= MAX_EXCLUDED_USERS) {
    filterExcludedUser += `AND A.user_id NOT IN (:excluded_users)`;
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->